lib/core.mli

```diff
@@ -536,56 +533,35 @@ val use_state : (unit -> 'state) -> 'state * (('state -> 'state) -> unit)
     let use_state initial = use_state_internal Imports.react initial]
 
 val use_reducer :
-  ('state -> 'action -> 'state) -> 'state -> 'state * ('action -> unit)
-  [@@js.custom
-    let (use_reducer_internal :
-             Imports.react
-          -> ('state -> 'action -> 'state)
-          -> 'state
-          -> 'state * ('action -> unit) ) =
-     fun (react : Imports.react) (reducer : 'state -> 'action -> 'state)
-         (init : 'state) ->
-      let result =
-        Ojs.call
-          (Imports.react_to_js react)
-          "useReducer"
-          [| Ojs.fun_to_js 2 (fun (state : Ojs.t) (action : Ojs.t) ->
-                 Obj.magic (reducer (Obj.magic state) (Obj.magic action)) )
-           ; Obj.magic init |]
-      in
-      (Obj.magic (Ojs.array_get result 0), Obj.magic (Ojs.array_get result 1))
-
-    let use_reducer reducer initial =
-      use_reducer_internal Imports.react reducer initial]
-
-val use_reducer_with_map_state :
-     ('state -> 'action -> 'state)
-  -> 'initialState
-  -> ('initialState -> 'state)
+     reducer:('state -> 'action -> 'state)
```

Comment:
Maybe we could put `reducer` last, and remove the label. Otherwise it's a bit strange to do `use_reducer ~reducer` 🤔

Reply:
Good idea. I was thinking you'd usually use a separate `reducer` function, which would make it equally repetitive anyway, and using a labelled argument would also help guide user's to naming that function. That's a very minor point though.